controllers/datamover.go

```diff
@@ -558,6 +559,59 @@ func (r *DPAReconciler) createResticSecretsPerBSL(dpa *oadpv1alpha1.DataProtecti
 	return nil, nil
 }
 
+func (r *DPAReconciler) createResticCustomCASecret(dpa *oadpv1alpha1.DataProtectionApplication, bsl velerov1.BackupStorageLocation) (*corev1.Secret, error) {
+	insecureSkipTLSVerify, skipPresent := bsl.Spec.Config["insecureSkipTLSVerify"]
+	if !skipPresent || insecureSkipTLSVerify != "false" {
+		return nil, nil
+	}
+	if bsl.Spec.ObjectStorage.CACert == nil {
+		return nil, errors.New("insecureSkipTLSVerify set to false with no caCert specified")
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-volsync-restic-customca", bsl.Name),
+			Namespace: bsl.Namespace,
+			Labels: map[string]string{
+				oadpv1alpha1.OadpBSLnameLabel:     bsl.Name,
+				oadpv1alpha1.OadpBSLProviderLabel: bsl.Spec.Provider,
+				oadpv1alpha1.OadpOperatorLabel:    "True",
+			},
+		},
+	}
+
+	mutateSecret := func() error {
+		err := controllerutil.SetControllerReference(dpa, secret, r.Scheme)
```

Comment:
I'm not sure if this will create an issue with the volsync controller. Just something to keep in mind.

Reply:
I copied this from the other secret creation code without fully understanding what it does. What kind of problems are you anticipating?

Reply:
My concern was that the volsync controller may have also been setting a controller reference on this secret: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#SetControllerReference
Although I just looked through their controller and do not see them doing so, which means this should be fine.